skimage/util/lazy.py

```diff
@@ -0,0 +1,65 @@
+import sys
+import importlib
+import importlib.util
+
+
+def install_lazy(module_name, submodules=None, submod_funcs=None):
+    """Install lazily loaded submodules and functions.
+
+    Parameters
+    ----------
+    module_name : str
+        Typically use __name__.
+    submodules : list
+        List of submodules to install.
+    submod_funcs : dict
+        Dictionary of submodule -> list of functions.  These functions are
+        imported as they are used.
+    """
+    if submod_funcs is None:
+        submod_funcs = {}
+
+    if submodules is None:
+        submodules = []
+
+    all_funcs = []
+    for mod, funcs in submod_funcs.items():
+        all_funcs.extend(funcs)
+
+    def __getattr__(name):
+        if name in submodules:
+            lazy_mod = require(f'skimage.{name}')
+            return lazy_mod
+        elif name in all_funcs:
+            for mod, funcs in submod_funcs.items():
```

Comment:
There should definitely be a reverse lookup table here so you don't have to execute a loop every time a new attribute is accessed. 

Something like:

```python
# On the call to ``install_lazy``
name_to_submod = {func: mod for mod, funcs in submod_funcs.items() for func in funcs}
```

Then use it as 

```python
elif name in name_to_submod :
    mod = name_to_submod[name]
    submod = importlib.import_module(f'{module_name}.{mod}')
    return getattr(submod, name)
```

Reply:
Done